src/V3AstNodes.cpp

```diff
@@ -1002,6 +1002,12 @@ const char* AstJumpBlock::broken() const {
     BROKEN_RTN(!labelp()->brokeExistsBelow());
     return nullptr;
 }
+bool AstJumpBlock::isPure() {
+    for (AstNode* stmtp = this->stmtsp(); stmtp; stmtp = stmtp->nextp()) {
+        if (!stmtp->isPure()) return false;
+    }
+    return true;
+}
```

Comment:
This is expensive to calculate, please add a   

  VIsCached m_purity;  // Pure state


member and use it like other node types with expensive pure calculations, thanks.

Reply:
Done.